frontend/src/utils/test/budgetUtil.test.ts

```diff
@@ -0,0 +1,135 @@
+      ['expense'],
+      22.03
+    );
+    expect(categoryAmountMap['FOOD / Pizza']).toHaveProperty(
+      ['limit'],
+      50.00
+    );
+    expect(categoryAmountMap['FOOD / Coffee / Starbucks']).toHaveProperty(
+      ['expense'],
+      0
+    );
+    expect(categoryAmountMap['FOOD / Coffee / Starbucks']).toHaveProperty(
+      ['limit'],
+      20.00
+    );
+    expect(categoryAmountMap[income]).toHaveProperty(['expense'], 0);
+    expect(categoryAmountMap[income]).toHaveProperty(['limit'], 0);
+    expect(categoryAmountMap[education]).toHaveProperty(['expense'], 78.54);
+    expect(categoryAmountMap[education]).toHaveProperty(['limit'], 200.00);
+    expect(categoryAmountMap[housing]).toHaveProperty(['expense'], 0);
+    expect(categoryAmountMap[housing]).toHaveProperty(['limit'], 1200.00);
+    expect(categoryAmountMap[utility]).toHaveProperty(['expense'], 252.5);
+    expect(categoryAmountMap[utility]).toHaveProperty(['limit'], 350.00);
+  });
+
+  it('should test generateParentCategoryWithdrawMaps', () => {
+    const parentCategoryWithdrawMaps = generateParentCategoryWithdrawMaps(transactions, categories);
+    
+    expect(Object.keys(parentCategoryWithdrawMaps)).toHaveLength(7);
+    expect(parentCategoryWithdrawMaps).toHaveProperty([food], 229.7);
+    expect(parentCategoryWithdrawMaps).toHaveProperty([transportation], 588);
+    expect(parentCategoryWithdrawMaps).toHaveProperty([education], 78.54);
+    expect(parentCategoryWithdrawMaps).toHaveProperty([others], 123.45);
+    expect(parentCategoryWithdrawMaps).toHaveProperty([entertainment], 21.23);
+    expect(parentCategoryWithdrawMaps).toHaveProperty([utility], 252.5);
+    expect(parentCategoryWithdrawMaps).toHaveProperty([income], 0);
+  });
+
+  it('should test sumDepositAndWithdraw', () => {
+    const transactionMonths = getTransactionMonths(transactions);
+    const depositWithdrawMaps = sumDepositAndWithdraw(transactions, transactionMonths);
+    
+    expect(Object.keys(depositWithdrawMaps)).toHaveLength(11);
+    expect(depositWithdrawMaps['2018-11']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-10']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-09']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-08']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-07']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-06']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-05']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-04']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-03']).toHaveProperty(['deposit'], 0);
+    expect(depositWithdrawMaps['2018-02']).toHaveProperty(['deposit'], 8642.24);
+    expect(depositWithdrawMaps['2018-01']).toHaveProperty(['deposit'], 8642.24);
+
+    // expect(depositWithdrawMaps['2018-11']).toHaveProperty(['withdraw'], 325.35);
```

Comment:
floating point issue. Need to figure out how to check this.

Reply:
Let's discuss this week.